src/client/pydaos/pydaos_shim.c

```diff
@@ -115,58 +122,109 @@ __shim_handle__err_to_str(PyObject *self, PyObject *args)
  */
 
 static PyObject *
-cont_open(int ret, uuid_t puuid, uuid_t cuuid, int flags)
+cont_open(int ret, char *pool, char *cont, int flags)
 {
-	PyObject	*return_list;
-	daos_handle_t	 coh = {0};
-	daos_handle_t	 poh = {0};
-	int		 rc;
+	PyObject			*return_list;
+	struct open_handle		*hdl = NULL;
+	daos_handle_t			coh = {0};
+	daos_handle_t			poh = {0};
+	daos_prop_t			*prop = NULL;
+	struct daos_prop_entry		*entry;
+	struct daos_prop_co_roots	*roots;
+	int				rc;
 
 	if (ret != DER_SUCCESS) {
 		rc = ret;
 		goto out;
 	}
 
 	/** Connect to pool */
-	rc = daos_pool_connect(puuid, "daos_server", DAOS_PC_RW, &poh,
+	rc = daos_pool_connect(pool, NULL, DAOS_PC_RW, &poh,
 			       NULL, NULL);
 	if (rc)
 		goto out;
 
 	/** Open container */
-	rc = daos_cont_open(poh, cuuid, DAOS_COO_RW, &coh, NULL, NULL);
+	rc = daos_cont_open(poh, cont, DAOS_COO_RW, &coh, NULL, NULL);
 	if (rc)
-		daos_pool_disconnect(poh, NULL);
+		goto out;
+
+	/** Retrieve container properties via cont_query() */
+	prop = daos_prop_alloc(0);
+	if (prop == NULL) {
+		rc = -ENOMEM;
+		goto out;
+	}
+
+	rc = daos_cont_query(coh, NULL, prop, NULL);
+	if (rc)
+		goto out;
+
+	/** Verify that this is a python container */
+	entry = daos_prop_entry_get(prop, DAOS_PROP_CO_LAYOUT_TYPE);
+	if (entry == NULL || entry->dpe_val != DAOS_PROP_CO_LAYOUT_PYTHON) {
+		D_ERROR("Container is not a python container\n");
+		rc = -DER_INVAL;
+		goto out;
```

Comment:
Should we use D_GOTO in this file?

Reply:
D_GOTO is supposed to be deprecated .... but we are not enforcing it yet ...

Reply:
I didn't realize. Why? And is this now the preferred style?

Reply:
D_GOTO was deprecated like 3 years ago, but its use has exponentially increased since then :-)